src/storage/store/rel_table.cpp

```diff
@@ -166,7 +168,30 @@ bool RelTable::scanInternal(Transaction* transaction, TableScanState& scanState)
     return scanState.scanNext(transaction);
 }
 
+void RelTable::checkRelMultiplicityConstraint(Transaction* transaction,
+    const TableInsertState& state) const {
+    const auto& insertState = state.constCast<RelTableInsertState>();
+    KU_ASSERT(insertState.srcNodeIDVector.state->getSelVector().getSelSize() == 1 &&
+              insertState.dstNodeIDVector.state->getSelVector().getSelSize() == 1);
+
+    const auto throwFunc = [](const std::string& tableName, offset_t nodeOffset,
+                               RelDataDirection direction) {
+        throw RuntimeException(ExceptionMessage::violateRelMultiplicityConstraint(tableName,
+            std::to_string(nodeOffset), RelDataDirectionUtils::relDirectionToString(direction)));
+    };
+    if (fwdRelTableData->getMultiplicity() == common::RelMultiplicity::ONE) {
+        throwIfNodeHasRels(transaction, common::RelDataDirection::FWD, &insertState.srcNodeIDVector,
+            throwFunc);
+    }
+    if (bwdRelTableData->getMultiplicity() == common::RelMultiplicity::ONE) {
+        throwIfNodeHasRels(transaction, common::RelDataDirection::BWD, &insertState.dstNodeIDVector,
+            throwFunc);
+    }
+}
+
 void RelTable::insert(Transaction* transaction, TableInsertState& insertState) {
+    checkRelMultiplicityConstraint(transaction, insertState);
```

Comment:
No need to worry about this for updates right? This should be the case as long as updates cannot modify the source/dest nodes of an edge.

Reply:
yep.